### PR TITLE
fix: orderbook calculation bugs, expose more data

### PR DESCRIPTION
**Pull request by tyleroooo**

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt [hunk 1]
```diff
@@ -510,43 +517,55 @@ internal class OrderbookProcessor(parser: ParserProtocol) : BaseProcessor(parser
     }
 
     fun group(orderbook: List<Any>?, grouping: Double): List<Any>? {
-        return if (orderbook != null && orderbook.size > 0) {
+        return if (!orderbook.isNullOrEmpty()) {
```
Comment (yogurtandjam) [inline src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt:520]: [discussion/suggestion] IMO this is one those functions that has enough math and logic where it makes sense to test separately. we should probably be actively testing edge cases and difficult to reason-about situations like stuff that's been commented here.

i tried to start a pattern with adding [the processor tests](https://github.com/dydxprotocol/v4-abacus/tree/main/src/commonTest/kotlin/exchange.dydx.abacus/processor) to the commonTest folder.

**Review by yogurtandjam — COMMENTED** [review]
looks good but would really appreciate some unit tests to illustrate any edge or confusing cases we're concerned with. i'm not as familiar with this part of the code and there's a lot of logic here so i'm hesitant to stamp

File: src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt [hunk 2]
```diff
@@ -510,43 +517,55 @@ internal class OrderbookProcessor(parser: ParserProtocol) : BaseProcessor(parser
...
+            // orderbook always ordered in increasing depth which is either increasing (asks) or decreasing (bids) price
+            // we want to round asks up and bids down so they don't have an overlapping group in the middle
             val firstPrice = parser.asDouble(parser.value(orderbook.firstOrNull(), "price"))!!
-            var floor = Rounder.round(firstPrice, grouping)
-            var ceiling = floor + grouping
+            val lastPrice = parser.asDouble(parser.value(orderbook.lastOrNull(), "price"))!!
```
[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt:524]: we don't know if these are bids or asks, right?

File: src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt (hunk 1 above)
Comment (tyleroooo) [reply src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt:520]: I think the existing orderbook tests hit the main points pretty well............and I feel like it would take me forever to set up new tests.

Comment (tyleroooo) [reply src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt:520]: Yeah the orderbook tests basically hit this directly

**Review by yogurtandjam — APPROVED** [review]
